### PR TITLE
Refactor job application employment history

### PR DESCRIPTION
**Pull request by cpjmcquillan**
## Jira ticket URL
https://dfedigital.atlassian.net/browse/TEVA-1905

## Changes in this PR
- Drop job application details
- Remove job application details component
- Create employments table for job applications
- Allow jobseekers to add employments to their job application (a la references)
- Add a date validator (which I think cleans up how we deal with and validate dates a little bit)

### CODE REVIEW

**Comment by martin-bangoura** [comment]
Review app deployed to <https://teaching-vacancies-review-pr-3202.london.cloudapps.digital>

File: config/locales/jobseekers.yml [hunk 1]
```diff
@@ -131,19 +131,21 @@ en:
...
-        employment_history: Role
-        form:
-          employment_history:
-            caption: Current role and employment history
-            heading: Add a role
-            save: Save role
-            title: Add a role - Current role and employment history - Application
-      employment_history:
-        current_role: Current role
+      employments:
```
Comment (alexbowen) [inline config/locales/jobseekers.yml:134]: employments is a strange way to call employment/work history imo

**Review by cesidio — APPROVED** [review]
Nice work, the only missing bit I guess is having these extra tables in analytics and analytics-pii YAML files

**Comment by cpjmcquillan** [comment]
> Nice work, the only missing bit I guess is having these extra tables in analytics and analytics-pii YAML files

Yeah I agree, I want to chat with Steven to understand more what we think is PII for all the job applications stuff

**Pull request merged**